test/exclude-members.test.js

```diff
@@ -154,3 +155,122 @@ test('add and remove a person, post on the new feed', async (t) => {
   await p(alice.close)(true)
   await p(bob.close)(true)
 })
+
+test("If you're not the excluder nor the excludee then you should still be in the group and have access to the new epoch", async (t) => {
+  // alice creates the group. adds bob and carol. removes bob.
+  // bob gets added and is removed
+  // carol stays in the group
+  const alice = Testbot({
+    keys: ssbKeys.generate(null, 'alice'),
+    mfSeed: Buffer.from(
+      '000000000000000000000000000000000000000000000000000000000000a1ce',
+      'hex'
+    ),
+  })
+  const bob = Testbot({
+    keys: ssbKeys.generate(null, 'bob'),
+    mfSeed: Buffer.from(
+      '0000000000000000000000000000000000000000000000000000000000000b0b',
+      'hex'
+    ),
+  })
+  const carol = Testbot({
+    keys: ssbKeys.generate(null, 'carol'),
+    mfSeed: Buffer.from(
+      '00000000000000000000000000000000000000000000000000000000000ca201',
+      'hex'
+    ),
+  })
+
+  await alice.tribes2.start()
+  await bob.tribes2.start()
+  await carol.tribes2.start()
+  t.pass('tribes2 started for everyone')
+
+  await p(alice.metafeeds.findOrCreate)()
+  const bobRoot = await p(bob.metafeeds.findOrCreate)()
+  const carolRoot = await p(carol.metafeeds.findOrCreate)()
+
+  await replicate(alice, bob)
+  await replicate(alice, carol)
+  await replicate(bob, carol)
+  t.pass('everyone replicates their trees')
+
+  const { id: groupId, writeKey: writeKey1 } = await alice.tribes2
+    .create()
+    .catch((err) => t.error(err, 'alice failed to create group'))
+
+  await replicate(alice, carol)
+
+  await alice.tribes2
+    .addMembers(groupId, [bobRoot.id, carolRoot.id])
+    .catch((err) => t.error(err, 'add bob fail'))
+
+  await replicate(alice, carol)
+
+  await carol.tribes2.acceptInvite(groupId)
+
+  await replicate(alice, carol)
+
+  const {
+    value: { author: firstFeedId },
+  } = await carol.tribes2
+    .publish({
+      type: 'test',
+      text: 'first post',
+      recps: [groupId],
+    })
+    .catch((err) => t.error(err, 'carol failed to publish on first feed'))
+
+  await replicate(alice, carol)
+
+  await alice.tribes2
+    .excludeMembers(groupId, [bobRoot.id])
+    .catch((err) => t.error(err, 'remove member fail'))
+
+  await replicate(alice, carol)
+
+  // TODO: maybe remove?
+  await p(setTimeout)(5000)
+
+  const {
+    value: { author: secondFeedId },
+  } = await carol.tribes2
+    .publish({
+      type: 'test',
+      text: 'second post',
+      recps: [groupId],
+    })
+    .catch(t.fail)
+
+  t.notEquals(
+    secondFeedId,
+    firstFeedId,
+    'feed for second publish is different to first publish'
+  )
```

Comment:
This fails.

Checked replication and carol seems to catch up to the latest message on the feed that the re-addition is published to by alice, but only alice seems to find the re-addition in start(). Does that mean that carol doesn't manage to decrypt the addition? Still happens with 10 seconds timeout before carol publishes.

Reply:
in box2/monitor.js we need to change it to something like

```js
      //pull.filter(isForeignLeafFeedBranch),
      pull.asyncMap((leafBranch, cb) => {
        if (!rootKeys) return cb()
        if (isForeignLeafFeedBranch(leafBranch)) {
          const theirRoot = leafBranch[0]
          const theirLeaf = leafBranch[3]
          ssb.metafeeds.getTree(rootKeys.id, (err, myRoot) => {
```
with an else where we do the reverse case

Reply:
andré described the problem like:

The problem was that: Carol tries to create a triangle with Alice as soon as Carol discover's Alice's epoch 0 feed, BUT at that point Carol doesn't yet have an epoch 0 feed, so creating triangle fails, and Carol never tries it again.

However, Carol SHOULD create a triangle with Alice's epoch 0 feed as soon as Carol creates her own epoch 0 feed

Reply:
We're putting this bug in the "fridge" for now by putting all add-member msgs on the additions feed, something we were thinking about anyway. TODO make issue etc

Reply:
Andre got a task to update the exclusion spec. Should make issue for this repo

Reply:
Rewrote this issue to be just about moving stuff to the additions feed. Will work on that now #79

Reply:
seems like this might work after merging #85 into this